daemon/containerd/image.go

```diff
@@ -25,17 +25,46 @@ import (
 
 var errInconsistentData error = errors.New("consistency error: data changed during operation, retry")
 
+type errPlatformNotFound struct {
+	wanted   ocispec.Platform
+	imageRef string
+}
+
+func (e *errPlatformNotFound) NotFound() {}
+func (e *errPlatformNotFound) Error() string {
+	msg := "image with reference " + e.imageRef + " was found but does not match the specified platform"
+	if e.wanted.OS != "" {
+		msg += ": wanted " + platforms.Format(e.wanted)
+	}
```

Comment:
Playing around a bit as I was wondering if putting this after a `:` delimiter may feel like it's a separate error that was wrapped. I guess we could included it between braces;

```
image with reference foo:bar was found but does not match the requested platform (linux/amd64)
```


Only downside of that would be if we'd have windows (` the requested platform (windows(10.0.17763)/amd64)`)

Reply:
The `wanted: ` part is also present in the graphdrivers error, see above comment.